master/buildbot/db/changes.py

```diff
@@ -66,27 +67,30 @@ def thd(conn):
             self.check_length(ch_tbl.c.repository, repository)
             self.check_length(ch_tbl.c.project, project)
 
-            r = conn.execute(ch_tbl.insert(), dict(
-                author=author,
-                comments=comments_val,
-                is_dir=is_dir,
-                branch=branch,
-                revision=revision,
-                revlink=revlink,
-                when_timestamp=datetime2epoch(when_timestamp),
-                category=category,
-                repository=repository,
-                codebase=codebase,
-                project=project))
-            changeid = r.inserted_primary_key[0]
+            changeid = None
+            with conn_execute(conn, ch_tbl.insert(), dict(
```

Comment:
Could you move this `dict` to a separate variable? Don't you think that now the `with` looks scary?